### PR TITLE
Added import from JSON feature

### PR DESCRIPTION
**Pull request by NickGeek**

### CODE REVIEW

File: src/styles/components/_dialog.scss [hunk 1]
```diff
@@ -26,3 +26,7 @@
   margin-bottom: 0.5em;
   padding-bottom: 0.5em;
 }
+
+.dialog button {
```
Comment (arkon) [inline src/styles/components/_dialog.scss]: Try to follow the [BEM naming convention](http://getbem.com/naming/). (Same below)

File: src/scripts/components/CRCMaker.js [hunk 2]
```diff
@@ -312,6 +325,18 @@ export default class CRCMaker extends Component {
           ) }
         </main>
 
+        { state.importVisible &&
+          <Dialog title='Import from JSON' onClose={this.toggleImport}>
+            <textarea id='json-import' onChange={e => this.inputJson = e.target.value}></textarea>
+
+            <button onClick={() => {
```
Comment (arkon) [inline src/scripts/components/CRCMaker.js]: Extract the logic into a function in the class. Also try to handle the case where the JSON might be improperly formatted.

File: src/scripts/components/CRCMaker.js [hunk 3]
```diff
@@ -254,14 +264,16 @@ export default class CRCMaker extends Component {
...
                 <button onClick={this.toggleExport}>Export</button>
                 { state.exportVisible &&
                   <Dialog title='Export JSON' onClose={this.toggleExport}>
-                    <pre id='text-export' className='syntax'
-                      dangerouslySetInnerHTML={{__html: syntaxHighlight(state.cards)}} />
-
                     <button className='copy' data-clipboard-target='#text-export'>Copy</button>
                     <button onClick={this.toggleExport}>Close</button>
+
+                    <pre id='text-export' className='syntax'
```
Comment (arkon) [inline src/scripts/components/CRCMaker.js]: Why was this moved?

File: src/scripts/components/CRCMaker.js [hunk 4]
```diff
@@ -254,14 +264,16 @@ export default class CRCMaker extends Component {
                   </Dialog>
                 }
 
+                <button onClick={this.toggleImport}>Import</button>
+
```
Comment (arkon) [inline src/scripts/components/CRCMaker.js]: I'd rather the import dialog be around here too, since the export dialog is here.

File: src/scripts/components/CRCMaker.js (hunk 3 above)
Comment (NickGeek) [reply src/scripts/components/CRCMaker.js]: Oh, sorry. That was another idea I had that I forgot I left in (I didn't like having to scroll down to select "copy".

I can remove that from the branch.

File: src/scripts/components/CRCMaker.js (hunk 4 above)
Comment (NickGeek) [reply src/scripts/components/CRCMaker.js]: I don't think that's possible because header is hidden on the initial load

File: src/scripts/components/CRCMaker.js (hunk 2 above)
Comment (NickGeek) [reply src/scripts/components/CRCMaker.js]: Yep good idea. I can put that in its own component.

**Comment by NickGeek** [comment]
@arkon I think I have fixed up all the things that were commented on, thanks for the feedback.

**Comment by NickGeek** [comment]
I'm still a little confused on BEM but I think I've made the changes that are wanted.
